Source/p_setup.c

```diff
@@ -1038,6 +1040,78 @@ static void P_CreateBlockMap(void)
 
 #endif // MBF_STRICT
 
+//
+// P_VerifyBlockMap
+//
+// haleyjd 03/04/10: do verification on validity of blockmap.
+//
+static boolean P_VerifyBlockMap(int count)
+{
+  int x, y;
+  int *maxoffs = blockmaplump + count;
+
+  skipblstart = true;
+
+  for(y = 0; y < bmapheight; y++)
+  {
+    for(x = 0; x < bmapwidth; x++)
+    {
+      int offset;
+      int *list, *tmplist;
+      int *blockoffset;
+
+      offset = y * bmapwidth + x;
+      blockoffset = blockmaplump + offset + 4;
+
+      // check that block offset is in bounds
+      if(blockoffset >= maxoffs)
+      {
+        fprintf(stderr, "P_VerifyBlockMap: block offset overflow\n");
+        return false;
+      }
+
+      offset = *blockoffset;
+
+      // check that list offset is in bounds
+      if(offset < 4 || offset >= count)
+      {
+        fprintf(stderr, "P_VerifyBlockMap: list offset overflow\n");
+        return false;
+      }
+
+      list   = blockmaplump + offset;
+
+      if (*list != 0)
+        skipblstart = false;
```

Comment:
Well, strictly speaking we only need this part of this function. It doesn't do that much useful stuff else.

Reply:
Plus, I want to avoid GPL-3 sneak in:
https://github.com/team-eternity/eternity/blob/912b90e5751fac434869fbd083b61c15b7f746c7/source/p_setup.cpp#L8